R/checkModels.R

```diff
@@ -1,74 +1,203 @@
 ### called by splitModels() and/or defineModel().
 ### Contains all functions that check certain aspects of models.
 
+### called by splitModels() ----------------------------------------------------
+
+checkPersonGroupsConsistency <- function(d){
+  d <- eatTools::makeDataFrame(d, verbose=FALSE)
+  ### Eintraege in erster Spalte muessen unique sein und duerfen keine missings enthalten
+  if(any(is.na(d[,1]))){
+    stop("Person identifier in first column of 'person.groups' has missing values.")}
+  if(length(d[,1]) != length(unique(d[,1]))){
+    stop("Person identifier in first column of 'person.groups' is not unique.")}
+  ### die naechsten checks erfolgen jeweils fuer alle weiteren spalten
+  chk1 <- lapply(colnames(d)[-1], FUN = function (x){
+    ### gruppierungsvariablen duerfen nicht konstant sein und keine fehlenden Werte haben
+    if(length(unique(d[,x])) == 1){
+      stop(paste0("Column '",x,"' of 'person.groups' is constant."))}
+    if(any(is.na(d[,x]))){
+      stop(paste0("Column '",x,"' of 'person.groups' has missing values."))}
+  })
+
+  # wenn nur eine Spalte wird diese als IDs angenommen
+  if(checkmate::test_data_frame(person.groups, ncols = 1)){
+    warning("person.groups contains just one column; this is treated as person ids", call. = FALSE)
+    person.groups$group <- all.persons.lab
+    all.persons <- FALSE
+  }
+  # person.groups auf Plausibilitaet checken
+  if(!is.null(person.groups)){
+    # hat erste Spalte mehr Elemente als alle anderen
+    len <- sapply(person.groups, function(x) length(unique(x)))
+    len.log <- len < len[1]
+    if(!all(len.log[-1])) warning(paste0("first column of person.groups might not contain person ids; please check\n(number of unique elements is smaller than in another column)"),
+                                  call. = FALSE)
+  }
+
+  return(d)
+}
```

Comment:
the function `splitModels()` allows the input `person.groups()` to be NULL. Hence, we must allow NULL input also for the `d` argument of `checkPersonGroupsConsistency()`. I would suggest to add 
```r
checkPersonGroupsConsistency <- function(d){
  if(is.null(d)) {return(d)}
  d <- eatTools::makeDataFrame(d, verbose=FALSE)
```

Reply:
I'm sorry, I've overlooked that 
```r
  if(!is.null(person.groups)){
    person.groups<- checkPersonGroupsConsistency(person.groups)}
```
already allows `person.groups` to be NULL. Please ignore the last message.